litmus-portal/authentication/pkg/user/repository.go

```diff
@@ -16,7 +17,7 @@ type Repository interface {
 	FindUser(user *entities.User) (*entities.User, error)
 	UpdatePassword(userPassword *entities.UserPassword, isAdminBeingReset bool) error
 	CreateUser(user *entities.User) (*entities.User, error)
-	UpdateUser(user *entities.User) (*entities.User, error)
+	UpdateUser(c *gin.Context, user *entities.User) (*entities.User, error)
```

Comment:
Passing the `*gin.Context` to the repository breaks the "Clean Architecture" principle the server follows as the repository shouldn't be aware of what the API framework does, in order to use the `uid` that is parsed from `JWT Token`, please call it directly at the handler, assign it to the user struct/password struct that is being passed to the repository. 

You can refer to the below snippet of code that achieves the same without passing context deeper into core logic.
 
https://github.com/litmuschaos/litmus/blob/b7243c9979d2e166fe472994c20e8fc4a7bc1d1d/litmus-portal/authentication/api/handlers/user_handlers.go#L112-L115